jerry-core/ecma/builtin-objects/ecma-builtin-object.c

```diff
@@ -251,11 +188,37 @@ ecma_builtin_object_object_set_prototype_of (ecma_value_t arg1, /**< routine's f
     return ecma_copy_value (arg1);
   }
 
-  /* 6. TODO: es2015-subset profile does not support having a different
-   * [[SetPrototypeOf]] internal method */
+  ecma_object_t *obj_p = ecma_get_object_from_value (arg1);
+  ecma_value_t status;
+
+  /* 5. */
+#if ENABLED (JERRY_ES2015_BUILTIN_PROXY)
+  if (ecma_object_is_proxy (obj_p))
+  {
+    status = ecma_proxy_object_set_prototype_of (obj_p, arg2);
+
+    if (ECMA_IS_VALUE_ERROR (status))
+    {
+      return status;
+    }
+  }
+  else
+  {
+#endif /* ENABLED (JERRY_ES2015_BUILTIN_PROXY) */
+    status = ecma_op_ordinary_object_set_prototype_of (obj_p, arg2);
+#if ENABLED (JERRY_ES2015_BUILTIN_PROXY)
+  }
+#endif /* ENABLED (JERRY_ES2015_BUILTIN_PROXY) */
+
+  if (ecma_is_value_false (status))
```

Comment:
Isn't this only possible if the `obj_p` is a proxy?

Reply:
No, it can happen when the target object is not extensible, or the new prototype object would cause cyclical prototype chain.